File: src/Around.Core/Entities/Copter.cs
using System.Collections.Generic;
using Around.Core.Enums;

namespace Around.Core.Entities
{
    public class Copter
    {
        public Copter()
        {
        }

        public int Id { get; set; }

        public string Name { get; set; }

        public Status Status { get; set; }

        public double Latitude { get; set; }

        public double Longitude { get; set; }

        public double CostPerMinute { get; set; }

        public int BrandId { get; set; }

        public double MaxSpeed { get; set; }

        public double MaxFlightHeight { get; set; }

        public Control Control { get; set; }

        public DroneType DroneType { get; set; }

        public int? FullCharacteristicsId { get; set; }

        public virtual Brand Brand { get; set; }

        public virtual List<Rent> Rents { get; set; }

        public virtual FullCharacteristics FullCharacteristics { get; set; }

        public Copter CreateFromDto(CopterAggregate copter)
        {
            Name = copter.Name;
            Status = (Status)copter.Status;
            Latitude = copter.Latitude;
            Longitude = copter.Longitude;
            CostPerMinute = copter.CostPerMinute;
            BrandId = copter.BrandId;
            MaxSpeed = copter.MaxSpeed;
            MaxFlightHeight = copter.MaxFlightHeight;
            Control = (Control)copter.Control;
            DroneType = (DroneType)copter.DroneType;

            return this;
        }

        public Copter Update(Copter newCopter)
        {
            Name = newCopter.Name;
            Status = newCopter.Status;
            Latitude = newCopter.Latitude;
            Longitude = newCopter.Longitude;
            CostPerMinute = newCopter.CostPerMinute;
            BrandId = newCopter.BrandId;
            MaxSpeed = newCopter.MaxSpeed;
            MaxFlightHeight = newCopter.MaxFlightHeight;
            Control = newCopter.Control;
            DroneType = newCopter.DroneType;

            return this;
        }
    }
}

File: src/Around.Frontend/Around-Frontend/src/app/model/copter.ts
import { CopterDto } from '../interface/copter-dto';

export class Copter {
    constructor(
        public id: number,
        public name: string,
        public status: string,
        public latidude: string,
        public longitude: string,
        public costPerMinute: number,
        public brandName: string,
        public maxSpeed: number,
        public maxFlightHeight: number,
        public control: string,
        public droneType: string) {
    }
    public static Create(dto: CopterDto): Copter {
        return new Copter(dto.id, dto.name, dto.status,
            dto.latidude, dto.longitude, dto.costPerMinute,
            dto.brandName, dto.maxSpeed, dto.maxFlightHeight,
            dto.control, dto.droneType);
    }
}
